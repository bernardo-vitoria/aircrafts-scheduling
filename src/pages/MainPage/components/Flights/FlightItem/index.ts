import FlightItem from "./FlightItem";

export { FlightItem };