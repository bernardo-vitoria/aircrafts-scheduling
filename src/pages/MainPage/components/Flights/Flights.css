.flights-body {
    overflow-x: hidden;
    overflow-y: auto;
    height: 84vh;
}

.flights-header {
    display: flex;
    justify-content: center;
    align-items: end;
}

.flights-header_title {
    padding: 8px;
    text-align: center;
    padding-bottom: 4px;
};

.flights-header_error {
    display: flex
};
