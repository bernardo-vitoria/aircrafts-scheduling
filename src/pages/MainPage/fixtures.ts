const aircraft = { ident: "GABCD", type: "A320", economySeats: 186, base: "EGKK" };

const aircrafts = [
  { ident: "GABCD", type: "A320", economySeats: 186, base: "EGKK" },
];

const flight = {
  id: "AS1001",
  departuretime: 21600,
  arrivaltime: 26100,
  readable_departure: "06:00",
  readable_arrival: "07:15",
  origin: "LFSB",
  destination: "LFMN",
};

const flights = [
  {
    id: "AS1001",
    departuretime: 21600,
    arrivaltime: 26100,
    readable_departure: "06:00",
    readable_arrival: "07:15",
    origin: "LFSB",
    destination: "LFMN",
  },
  {
    id: "AS1002",
    departuretime: 27900,
    arrivaltime: 32100,
    readable_departure: "07:45",
    readable_arrival: "08:55",
    origin: "LFMN",
    destination: "LFSB",
  },
  {
    id: "AS1025",
    departuretime: 22800,
    arrivaltime: 28200,
    readable_departure: "06:20",
    readable_arrival: "07:50",
    origin: "LFSB",
    destination: "EDDH",
  },
  {
    id: "AS1026",
    departuretime: 30000,
    arrivaltime: 35100,
    readable_departure: "08:20",
    readable_arrival: "09:45",
    origin: "EDDH",
    destination: "LFSB",
  },
];

export {
    aircraft,
    aircrafts,
    flight,
    flights
};
