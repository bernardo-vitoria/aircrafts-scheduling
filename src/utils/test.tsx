// @ts-nocheck
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { I18nextProvider } from "react-i18next";
import { QueryClient, QueryClientProvider } from "react-query";
import { MemoryRouter } from "react-router-dom"; // 4.0.0
import { queries } from "@testing-library/react";
import {
  render,
  RenderResult,
} from "@testing-library/react";
import i18n from "../config/i18nForTests";

const customRender = (ui: React.ReactElement, options = {}): RenderResult => {
  const CombinedProviders: React.FC = ({ children }) => {
    const config = {
      queries: {
        retry: false,
        cacheTime: 0,
        refetchOnWindowFocus: false,
        retryDelay: 0,
      },
    };

    const queryClient = new QueryClient({ defaultOptions: config });

    return (
        <QueryClientProvider client={queryClient}>
          <I18nextProvider i18n={i18n}>
            <MemoryRouter>{children}</MemoryRouter>
          </I18nextProvider>
        </QueryClientProvider>
    );
  };

  return render(ui, { wrapper: CombinedProviders, queries, ...options });
};

export {
  customRender as render
};