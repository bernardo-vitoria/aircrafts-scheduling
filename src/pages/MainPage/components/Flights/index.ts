import Flights from './Flights';

export { Flights };