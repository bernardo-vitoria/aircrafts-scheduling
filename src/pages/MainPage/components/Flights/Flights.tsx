import React, { useState } from 'react';
import { FlightItem } from './FlightItem';
import { Paper, ErrorIcon, Tooltip, Stack } from '../../../../components';
import { Flight } from '../../types';
import { useTranslation } from 'react-i18next';
import "./Flights.css";

type Props = {
    flights: Flight[];
};

const Flights: React.FC<Props> = ({flights}) => {
  const [t] = useTranslation();
  const [hasError, setHasError] = useState(false);

    return (
      <Paper sx={{ height: "90vh", minWidth: "211px" }}>
        <div className="flights-header">
          <div className="flights-header_title">
            <b>{t("mainPage:flights.flights")}</b>
          </div>
          <div className="flights-header_error">
            {hasError && (
              <Tooltip describeChild title={`${t("mainPage:flights.error")}`}>
                <ErrorIcon sx={{ color: "#e63946" }} />
              </Tooltip>
            )}
          </div>
        </div>
        <Stack
          direction="column"
          justifyContent="space-evenly"
          alignItems="center"
          spacing={2}
        >
          <div className="flights-body">
            {flights.map((flight: Flight) => (
              <FlightItem
                flight={flight}
                key={`flightItem-${flight.id}`}
                hasError={hasError}
                setHasError={setHasError}
              />
            ))}
          </div>
        </Stack>
      </Paper>
    );
};

export default Flights;