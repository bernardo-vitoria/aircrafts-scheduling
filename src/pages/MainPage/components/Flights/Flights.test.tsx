import Flights from "./Flights";
import { render } from "../../../../utils/test";
import { flights } from "../../fixtures";

describe("Flights", () => {
  it("should match snapshot", () => {
    const { container } = render(<Flights flights={flights} />);

    expect(container).toMatchSnapshot();
  });
});
