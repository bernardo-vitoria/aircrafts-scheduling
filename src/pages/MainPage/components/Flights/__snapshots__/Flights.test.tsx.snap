// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flights should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1v9y39d-MuiPaper-root"
  >
    <div
      class="flights-header"
    >
      <div
        class="flights-header_title"
      >
        <b>
          Flights
        </b>
      </div>
      <div
        class="flights-header_error"
      />
    </div>
    <div
      class="css-1lgskws-MuiStack-root"
    >
      <div
        class="flights-body"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-poj73t-MuiPaper-root"
        >
          <span
            class="flight-item"
          >
            <div>
              <b>
                AS1001
              </b>
            </div>
            <div
              class="card-body"
            >
              <div
                class="card-body_column"
              >
                <div>
                  LFSB
                </div>
                <div>
                  06:00
                </div>
              </div>
              <div
                class="card-body_column"
              >
                <div>
                  LFMN
                </div>
                <div>
                  07:15
                </div>
              </div>
            </div>
          </span>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-poj73t-MuiPaper-root"
        >
          <span
            class="flight-item"
          >
            <div>
              <b>
                AS1002
              </b>
            </div>
            <div
              class="card-body"
            >
              <div
                class="card-body_column"
              >
                <div>
                  LFMN
                </div>
                <div>
                  07:45
                </div>
              </div>
              <div
                class="card-body_column"
              >
                <div>
                  LFSB
                </div>
                <div>
                  08:55
                </div>
              </div>
            </div>
          </span>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-poj73t-MuiPaper-root"
        >
          <span
            class="flight-item"
          >
            <div>
              <b>
                AS1025
              </b>
            </div>
            <div
              class="card-body"
            >
              <div
                class="card-body_column"
              >
                <div>
                  LFSB
                </div>
                <div>
                  06:20
                </div>
              </div>
              <div
                class="card-body_column"
              >
                <div>
                  EDDH
                </div>
                <div>
                  07:50
                </div>
              </div>
            </div>
          </span>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-poj73t-MuiPaper-root"
        >
          <span
            class="flight-item"
          >
            <div>
              <b>
                AS1026
              </b>
            </div>
            <div
              class="card-body"
            >
              <div
                class="card-body_column"
              >
                <div>
                  EDDH
                </div>
                <div>
                  08:20
                </div>
              </div>
              <div
                class="card-body_column"
              >
                <div>
                  LFSB
                </div>
                <div>
                  09:45
                </div>
              </div>
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
